ui/packages/components/src/RunsPage/RunsPage.tsx

```diff
@@ -212,10 +226,20 @@ export function RunsPage({
     [cancelRun, getRun, getTraceResult, getTrigger, pathCreator, pollInterval, rerun]
   );
 
-  const options: Option[] = columns.map((column) => ({
-    id: column.accessorKey,
-    name: column.header?.toString() || column.accessorKey,
-  }));
+  const options = useMemo(() => {
+    const out = [];
+    for (const column of columns) {
+      if (!isColumnID(column.id)) {
+        continue;
+      }
```

Comment:
**question (non-blocking**: I'm guessing this logic is just for type safety? Part of TS that I don't love 😆

Reply:
TypeScript is warning about a real thing here. TanStack Table columns aren't required to have IDs. We _should_ always specify them, but technically they're optional

Reply:
Thanks - makes a lot of sense!